Tribler/Core/APIImplementation/LaunchManyCore.py

```diff
@@ -594,31 +595,33 @@ def early_shutdown(self):
         """ Called as soon as Session shutdown is initiated. Used to start
         shutdown tasks that takes some time and that can run in parallel
         to checkpointing, etc.
+        :returns a Deferred that will fire once all dependencies acknowledge they have shutdown.
         """
         self._logger.info("tlm: early_shutdown")
 
         self.cancel_all_pending_tasks()
 
         # Note: sesslock not held
+        wait_defer_list = []
         self.shutdownstarttime = timemod.time()
         if self.torrent_checker:
-            self.torrent_checker.shutdown()
+            wait_defer_list.append(self.torrent_checker.shutdown())
```

Comment:
do the @inlineCallbacks thing and create an issue for the `new gui` milestone so we can come back at this when the apocalyptic session overhaul comes.

I don't really like using that, but I can't think of a less ugly way of doing it and all this stuff is going to get rewritten soon anyway.